core/unit_test/TestRange.hpp

```diff
@@ -327,6 +327,71 @@ TEST(TEST_CATEGORY, range_reduce) {
   }
 }
 
+template <typename ExecSpace, typename StaticBatchSize>
+struct TestStaticBatchSize {
+  using value_type = int;  ///< alias required for the parallel_reduce
+  using view_type  = Kokkos::View<value_type *, ExecSpace>;
+
+  view_type m_flags;
+  view_type result_view;
+
+  struct AtomicAddTag {};
+  struct VerifyAtomicAddTag {};
+
+  int N;
+
+  TestStaticBatchSize(const size_t N_)
+      : m_flags(Kokkos::view_alloc(Kokkos::WithoutInitializing, "flags"), N_),
+        result_view(Kokkos::view_alloc(Kokkos::WithoutInitializing, "results"),
+                    N_),
+        N(N_) {}
+
+  void test_batch_size() {
+    typename view_type::HostMirror host_flags =
+        Kokkos::create_mirror_view(m_flags);
+
+    // Initialize the flags
+    Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, N), *this);
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecSpace, AtomicAddTag, StaticBatchSize>(0, N),
+        *this);
+
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecSpace, VerifyAtomicAddTag>(0, N), *this);
+
+    Kokkos::deep_copy(host_flags, m_flags);
+
+    int error_count = 0;
+    for (int i = 0; i < N; ++i) {
+      if (host_flags(i) != 1) {
+        ++error_count;
+      }
+    }
+    ASSERT_EQ(error_count, 0);
```

Comment:
I'm not sure to understand why you want to check that each flag is set to 1 on both host and device.

It seems the following is sufficient and shorted (not tested but you get the idea)
```suggestion
    bool success = false;
    Kokkos::parallel_reduce(
        Kokkos::RangePolicy<ExecSpace, VerifyTag>(0, N),
        *this
        Kokkos::LAnd<bool>(success)
    );
    ASSERT_TRUE(success);

    ...

KOKKOS_FUNCTION
void operator()(const VerifyTag, bool& current) const {
    current = m_flags(i) == 1 && current;
}
```